airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/parquet_parser.py

```diff
@@ -2,21 +2,130 @@
+            }
+
+        # Convert duration to seconds, then convert to the appropriate unit
+        if pa.types.is_duration(parquet_value.type):
+            duration = parquet_value.as_py()
+            duration_seconds = duration.total_seconds()
+            if parquet_value.type.unit == "s":
+                return duration_seconds
+            elif parquet_value.type.unit == "ms":
+                return duration_seconds * 1000
+            elif parquet_value.type.unit == "us":
+                return duration_seconds * 1_000_000
+            elif parquet_value.type.unit == "ns":
+                return duration_seconds * 1_000_000_000 + duration.nanoseconds
+            else:
+                raise ValueError(f"Unknown duration unit: {parquet_value.type.unit}")
+        else:
+            return parquet_value.as_py()
+
+    @staticmethod
+    def parquet_type_to_schema_type(parquet_type: pa.DataType) -> Mapping[str, str]:
+        """
+        Convert a pyarrow data type to an Airbyte schema type.
+        Parquet data types are defined at https://arrow.apache.org/docs/python/api/datatypes.html
+        """
+
+        if pa.types.is_timestamp(parquet_type):
+            return {"type": "string", "format": "date-time"}
+        elif pa.types.is_date(parquet_type):
+            return {"type": "string", "format": "date"}
+        elif ParquetParser._is_string(parquet_type):
+            return {"type": "string"}
+        elif pa.types.is_boolean(parquet_type):
+            return {"type": "boolean"}
+        elif ParquetParser._is_integer(parquet_type):
+            return {"type": "integer"}
+        elif pa.types.is_floating(parquet_type):
+            return {"type": "number"}
+        elif ParquetParser._is_object(parquet_type):
+            return {"type": "object"}
+        elif ParquetParser._is_list(parquet_type):
+            return {"type": "array"}
+        else:
+            raise ValueError(f"Unsupported parquet type: {parquet_type}")
+
+    @staticmethod
+    def _is_binary(parquet_type: pa.DataType) -> bool:
+        return bool(pa.types.is_binary(parquet_type) or pa.types.is_large_binary(parquet_type))
+
+    @staticmethod
+    def _is_integer(parquet_type: pa.DataType) -> bool:
+        return bool(pa.types.is_integer(parquet_type) or pa.types.is_duration(parquet_type))
+
+    @staticmethod
+    def _is_string(parquet_type: pa.DataType) -> bool:
+        return bool(
+            pa.types.is_time(parquet_type)
+            or pa.types.is_string(parquet_type)
+            or pa.types.is_large_string(parquet_type)
+            or pa.types.is_decimal(parquet_type)  # Return as a string to ensure no precision is lost
+            or ParquetParser._is_binary(parquet_type)  # Best we can do is return as a string since we do not support binary
+        )
+
+    @staticmethod
+    def _is_object(parquet_type: pa.DataType) -> bool:
+        return bool(pa.types.is_dictionary(parquet_type) or pa.types.is_struct(parquet_type))
```

Comment:
I noticed that most of these `_is...` method are all cast/wrapped into a boolean. Why do we need to cast these statements? I would think they already are booleanss

Reply:
mypy doesn't know about pyarrow so it doesn't know the method returns a bool. alternatives include
1.  add `# type: ignore` the lines so it doesn't complain
2. install stubs describing the pyarrow types

(2) sounds pretty involved for very little benefit since the pyarrow classes are only used here. I don't have a strong opinion between casting and telling mypy to ignore the "type error"